modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/DependencyNode.java

```diff
@@ -14,10 +21,12 @@
 public class DependencyNode {
 
   private String componentName;
+  private ComponentIdentifier componentIdentifier;
   private final Type type;
 
-  public DependencyNode(String componentName, Type type) {
+  public DependencyNode(String componentName, ComponentIdentifier componentIdentifier, Type type) {
```

Comment:
componentIdentifier -> Set it as Optional so users know the parameter may be omitted and also put it last in the list of parameters